### ISSUE TITLE
Handle unknown attributes and unknown types

### ISSUE THREAD

**Issue by hunterhector** [asyml/forte#435]
This issue is blocked by https://github.com/asyml/forte/issues/405.

**Is your feature request related to a problem? Please describe.**
Forte ontology defines the data types that can be used in the system,  sometimes the ontology evolves by adding new classes or new attributes. Some existing pipelines will break (during deserialization) since they encountered unknown types and attributes. 

Example:
The old ontology may define an `EntityMention`, with attribute `ner_type`. At some point we updated the definition, adding a new attribute, say `score` to `EntityMention`, now the `EntityMention` class changes from

```
class EntityMention:
   ner_type = ""
```
to 

```
class EntityMention:
   ner_type = ""
   score = 0
```

Loading data with the new types or attributes will break the deserialization. 

**Describe the solution you'd like**
Forte can be robust to these changes when a boolean flag is set to `on`, in such cases, it should ignore unknown entry types. 

On the other direction, when we load old data to a new pipeline, we can assign default `None` values to unknown entry attributes. 

**Describe alternatives you've considered**

**Additional context**

**Comment by J007X** [comment]
After some research related to the current version changes it looks like current introduction of version to our pre-defined classes are still not yet able to handle the dynamic nature of the ontology type changes. Maybe in the future some design changes in ontology that each type defined will somehow keep some kind of version or strict type definition

**Comment by hunterhector** [comment]
> After some research related to the current version changes it looks like current introduction of version to our pre-defined classes are still not yet able to handle the dynamic nature of the ontology type changes. Maybe in the future some design changes in ontology that each type defined will somehow keep some kind of version or strict type definition

Yes. Actually, the purpose of the current version system is not intended to solve these issues. We can consider revisiting this issue after the "data_efficiency" project, where we will introduce new data format and serialization mechanism

**Comment by J007X** [comment]
After reviewing code and take into consideration of previous discussions,  as the current jsonpickle does not have a usable (or very consistent) exception for us to utilize, and adding strict types are consider some (bigger and) new enhancement not only to this issue (as it might also solve #434). We can close this for now until new feature design can be settled (for example in the new data format and serialization mech mentioned above).

**Comment by hunterhector** [comment]
If we simply close this one, it is a bit hard to track. We probably should keep this open (I've already added a `won't fix` tag). Otherwise we need a different issue.

**Comment by J007X** [comment]
Thanks Hector for adding the tag. I'm creating a new feature request for this one and related 434 about "Allowing more strict type "enforcement"
